### PR TITLE
Fix ci build

### PR DESCRIPTION
**Pull request by mtekel**
Using updated pcap-testflight image, bump ruby spec to match image version. Bump acceptance tests golang. Address issues highlighted by new lint version.

Changes can be viewed commit-by-commit for more focused/limited scope of modifications.

### CODE REVIEW

File: src/pcap/test/bosh_test.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func TestNewBoshResolver(t *testing.T) {
 func TestAuthenticate(t *testing.T) {
 	bar, _, _, err := mock.NewResolverWithMockBoshAPI(nil)
 	if err != nil {
-		t.Errorf(err.Error())
+		t.Errorf("%s", err.Error())
```
Comment (peanball) [inline src/pcap/test/bosh_test.go:77]:
```suggestion
		t.Error(err)
```

File: src/pcap/api_test.go [hunk 2]
```diff
@@ -306,7 +306,7 @@ func TestCapture(t *testing.T) {
 				t.Errorf("capture() unexpected error during api creation: %v", err)
 			}
 
-			var connectToTargetFn = func(ctx context.Context, req *CaptureOptions, target AgentEndpoint, creds credentials.TransportCredentials, log *zap.Logger) (captureStream, error) {
+			var connectToTargetFn = func(ctx context.Context, req *CaptureOptions, target AgentEndpoint, creds credentials.TransportCredentials, log *zap.Logger) (captureStream, error) { //nolint:revive //keep vars even if unused for better context
 				return tt.stream, tt.err
```
Comment (peanball) [inline src/pcap/api_test.go:310]: I would argue based on the type you could still figure out what is put there. The name is not strictly needed.

File: src/pcap/test/integration/helpers.go [hunk 3]
```diff
@@ -176,17 +176,17 @@ func generateCerts(commonName string, dir string) (string, string, string, error
 	certPath := path.Join(dir, "cert.pem")
 	keyPath := path.Join(dir, "private.key")
 	caPath := path.Join(dir, "ca.pem")
-	err = os.WriteFile(certPath, certPEM.Bytes(), os.ModePerm)
+	err = os.WriteFile(certPath, certPEM.Bytes(), 0600)
 	if err != nil {
```
Comment (peanball) [inline src/pcap/test/integration/helpers.go:180]: I would prefer defining a variable with a name that describes the permissions, then use that. A 'magic number' linter should have noted this.

File: src/pcap/agent.go [hunk 4]
```diff
@@ -115,7 +115,7 @@ func (a *Agent) Capture(stream Agent_CaptureServer) (err error) {
 		return errorf(codes.InvalidArgument, "%w", err)
 	}
 
-	opts := req.Payload.(*AgentRequest_Start).Start.Capture
+	opts := req.Payload.(*AgentRequest_Start).Start.Capture //nolint:errcheck //this only returns one value
```
Comment (peanball) [inline src/pcap/agent.go:118]: we could have a check here for the cast.
```suggestion
	var opts *CaptureOptions
	if start, ok := req.Payload.(*AgentRequest_Start); ok {
		opts = start.Start.Capture
	} else {
		return errorf(codes.InvalidArgument, "invalid message: expected Payload of type StartAgentCapture")
	}
```
(not tested)

**Review by peanball — CHANGES_REQUESTED** [review]

**Comment by peanball** [comment]
some minor suggestions

File: src/pcap/api_test.go (hunk 2 above)
Comment (mtekel) [reply src/pcap/api_test.go:310]: For some reason original author decided to do it this way, so I'll leave it be.

File: src/pcap/agent.go (hunk 4 above)
Comment (peanball) [reply src/pcap/agent.go:118]: Ok. Maybe we can eventually return the cast result, but not in this small linter fix

**Review by peanball — APPROVED** [review]
looks good overall. approving for CI and we can merge if all of it passes :-)

**Pull request merged**